editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -785,6 +786,42 @@ void SpriteFramesEditor::_animation_add() {
 	animations->grab_focus();
 }
 
+void SpriteFramesEditor::_animation_duplicate() {
+	if (!frames->has_animation(edited_anim)) {
+		return;
+	}
+
+	int counter = 1;
+	String name = String(edited_anim) + " " + itos(counter);
+	while (frames->has_animation(name)) {
+		counter++;
+		name = String(edited_anim) + " " + itos(counter);
+	}
+	
+	undo_redo->create_action(TTR("Duplicate Animation"));
+	undo_redo->add_do_method(frames, "add_animation", name);	
+	for (int i = 0; i < frames->get_frame_count(edited_anim); i++) {
+		undo_redo->add_do_method(frames, "add_frame", name, frames->get_frame(edited_anim, i));
+	}
+	undo_redo->add_undo_method(frames, "remove_animation", name);	
+	undo_redo->add_do_method(this, "_update_library");
+	undo_redo->add_undo_method(this, "_update_library");
+
+	// Not sure whether this part is needed/wanted (`_add_animation` is doing the same thing).
```

Comment:
With the method name spelled wrongly it's a kinda misleading comment (wondering who suggested it like that...:laughing:). 
```suggestion
	// Not sure whether this part is needed/wanted (`_animation_add` is doing the same thing).
```